pkg/transaction/eventgenerator.go

```diff
@@ -66,23 +84,58 @@ func (e *Generator) CreateEvent(logEvent LogEvent, eventTime time.Time, metaData
 
 	eventData, err := e.createEventData(serializedLogEvent, eventFields)
 	if err != nil {
-		return
+		return event, err
 	}
 
-	event = beat.Event{
+	return beat.Event{
 		Timestamp: eventTime,
 		Meta:      metaData,
 		Private:   privateData,
 		Fields:    eventData,
+	}, nil
+}
+
+// CreateTransactionEvents - Creates new events to be sent to Condor
+func (e *Generator) CreateTransactionEvents(summaryEvent LogEvent, detailEvents []LogEvent, eventTime time.Time, metaData common.MapStr, eventFields common.MapStr, privateData interface{}) ([]beat.Event, error) {
```

Comment:
The method name sounds a bit obscure as it generates beats event wrapping both transaction summary and transaction events and not only transaction events. Can we  generalize the name to may be CreateEvents /CreateSampledEvents or something similar?

Reply:
CreateEvents, done